content/sips/sip-399.md

```diff
@@ -0,0 +1,103 @@
+
+## Motivation
+
+The motivation behind this proposal comes from requests by Perps integrators and partners who seek a better user experience when closing Perps positions. The current process can be cumbersome, requiring multiple steps and transactions. This flash loan utility contract aims to streamline the process, allowing users to close positions atomically, thereby improving efficiency and user satisfaction.
+
+## Specification
+
+### Overview
+
+The `PerpsFlashLoanUtil` contract will allow users to close their Perps positions using flash loans from the Aave protocol. The process involves wrapping USDC to snxUSD, repaying debt, modifying collateral, unwrapping to the original collateral type, swapping to USDC, and repaying the flash loan.
+
+### Rationale
+
+The proposed solution leverages existing Synthetix and Uniswap contracts to facilitate the flash loan process. By using a flash loan, users can close their Perps positions without needing to front the necessary collateral. The contract allows for dynamic setting of Uniswap pool fees and supporting various collateral types.
+
+### Technical Specification
+
+#### Interfaces
+
+```solidity
+/**
+ * @title Interface for the PerpsFlashLoanUtil contract.
+ * @notice Provides functions for requesting flash loans to close perps V3 positions.
+ */
+interface IPerpsFlashLoanUtil {
+    /**
+     * @notice Requests a flash loan.
+     * @param _amount The amount of USDC to flash loan.
+     * @param _collateralType The address of the collateral type to use.
+     * @param _marketId The ID of the market for the flash loan.
+     * @param _accountId The account ID to use for the flash loan.
+     */
+    function requestFlashLoan(
+        uint256 _amount,
+        address _collateralType,
+        uint128 _marketId,
+        uint128 _accountId
+    ) external;
+
+    /**
+     * @notice Executes the flash loan operation.
+     * @param asset The address of the asset being flash loaned.
+     * @param amount The amount of the asset being flash loaned.
+     * @param premium The premium to be paid for the flash loan.
+     * @param initiator The address that initiated the flash loan.
+     * @param params Additional parameters for the flash loan operation.
+     * @return success True if the operation was successful, false otherwise.
+     */
+    function executeOperation(
+        address asset,
+        uint256 amount,
+        uint256 premium,
+        address initiator,
+        bytes calldata params
+    ) external returns (bool success);
+
+    /**
+     * @notice Sets the Uniswap pool fee for a given collateral type.
+     * @param _collateralType The address of the collateral type.
+     * @param _poolFee The pool fee to be set for the collateral type.
+     */
+    function setPoolFee(address _collateralType, uint24 _poolFee) external;
```

Comment:
wouldn't it be better if the user can specify the address of the pool, as mentioned yesterday, in order to avoid having mroe configurations done by the protocol? Like he can route though any address, as long as he provides the address of the pool in the calldata?

Reply:
yes, updated 👍